tensorflow/compiler/xla/service/algebraic_simplifier_test.cc

```diff
@@ -5457,5 +5457,30 @@ TEST_F(AlgebraicSimplifierTest, CompareSame) {
               GmockMatch(m::Broadcast(m::ConstantScalar(true))));
 }
 
+TEST_F(AlgebraicSimplifierTest, CanDisableDotToMultiplyRewrite) {
+  // Some backends may have better performance by treating an outer product as a
+  // Dot, rather than a broadcast Multiply
+  const char* kModuleStr = R"(
+    HloModule m
+    test {
+      param1 = f32[64] parameter(0)
+      param2 = f32[64] parameter(1)
+      ROOT compare = f32[64, 64] dot(param1, param2),
+        lhs_contracting_dims={}, rhs_contracting_dims={}
+    })";
+
+  // Verify that the default is to re-write
+  TF_ASSERT_OK_AND_ASSIGN(auto m1, ParseAndReturnVerifiedModule(kModuleStr));
```

Comment:
ok

Reply:
in retrospect i don't agree with your request.  there are no other examples where this is done, including in tests which have been added to this file since this pull request was opened.  See `RemainderOfNPlusIotaOverflow` or `RepeatedRemainder` for examples.